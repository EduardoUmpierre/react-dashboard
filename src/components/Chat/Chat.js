import React from 'react'

const chat = props => <div>Chat</div>

export default chat
