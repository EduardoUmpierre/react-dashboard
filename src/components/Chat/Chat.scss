.chat {
    max-height: 18.750rem;
    overflow-y: auto;
    padding: 0.313rem 1.250rem;
}