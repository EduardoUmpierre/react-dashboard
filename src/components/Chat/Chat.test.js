import React from 'react'
import { shallow } from 'enzyme'
import Chat from './Chat'

describe('<Chat />', () => {
    const messages = [
        {
            displayPortraitLeft: false,
            message: 'Test',
            portrait: null,
            time: '2 min ago',
            userName: 'Eu'
        },
        {
            displayPortraitLeft: false,
            message: 'Test',
            portrait: null,
            time: '1 min ago',
            userName: 'Eu'
        }
    ]

    it('renders correctly', () => {
        const component = shallow(<Chat messages={messages} />)

        expect(component).toMatchSnapshot()
    })
})
