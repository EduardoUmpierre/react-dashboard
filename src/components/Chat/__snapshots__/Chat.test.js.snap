// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chat /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <chat
    messages={
      Array [
        Object {
          "displayPortraitLeft": false,
          "message": "Test",
          "portrait": null,
          "time": "2 min ago",
          "userName": "Eu",
        },
        Object {
          "displayPortraitLeft": false,
          "message": "Test",
          "portrait": null,
          "time": "1 min ago",
          "userName": "Eu",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="chat"
        >
          <loader
            loaded={
              Array [
                Object {
                  "displayPortraitLeft": false,
                  "message": "Test",
                  "portrait": null,
                  "time": "2 min ago",
                  "userName": "Eu",
                },
                Object {
                  "displayPortraitLeft": false,
                  "message": "Test",
                  "portrait": null,
                  "time": "1 min ago",
                  "userName": "Eu",
                },
              ]
            }
          >
            <chatItem
              displayPortraitLeft={false}
              message="Test"
              portrait={null}
              time="2 min ago"
              userName="Eu"
            />
            <chatItem
              displayPortraitLeft={false}
              message="Test"
              portrait={null}
              time="1 min ago"
              userName="Eu"
            />
          </loader>
        </div>,
        <chatForm
          isDisabled={false}
        />,
      ],
      "icon": "comments",
      "title": "Chat",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <loader
            loaded={
              Array [
                Object {
                  "displayPortraitLeft": false,
                  "message": "Test",
                  "portrait": null,
                  "time": "2 min ago",
                  "userName": "Eu",
                },
                Object {
                  "displayPortraitLeft": false,
                  "message": "Test",
                  "portrait": null,
                  "time": "1 min ago",
                  "userName": "Eu",
                },
              ]
            }
          >
            <chatItem
              displayPortraitLeft={false}
              message="Test"
              portrait={null}
              time="2 min ago"
              userName="Eu"
            />
            <chatItem
              displayPortraitLeft={false}
              message="Test"
              portrait={null}
              time="1 min ago"
              userName="Eu"
            />
          </loader>,
          "className": "chat",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <chatItem
                displayPortraitLeft={false}
                message="Test"
                portrait={null}
                time="2 min ago"
                userName="Eu"
              />,
              <chatItem
                displayPortraitLeft={false}
                message="Test"
                portrait={null}
                time="1 min ago"
                userName="Eu"
              />,
            ],
            "loaded": Array [
              Object {
                "displayPortraitLeft": false,
                "message": "Test",
                "portrait": null,
                "time": "2 min ago",
                "userName": "Eu",
              },
              Object {
                "displayPortraitLeft": false,
                "message": "Test",
                "portrait": null,
                "time": "1 min ago",
                "userName": "Eu",
              },
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "displayPortraitLeft": false,
                "message": "Test",
                "portrait": null,
                "time": "2 min ago",
                "userName": "Eu",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "displayPortraitLeft": false,
                "message": "Test",
                "portrait": null,
                "time": "1 min ago",
                "userName": "Eu",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "inputValue": undefined,
          "isDisabled": false,
          "onFormSubmit": undefined,
          "onInputChange": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="chat"
          >
            <loader
              loaded={
                Array [
                  Object {
                    "displayPortraitLeft": false,
                    "message": "Test",
                    "portrait": null,
                    "time": "2 min ago",
                    "userName": "Eu",
                  },
                  Object {
                    "displayPortraitLeft": false,
                    "message": "Test",
                    "portrait": null,
                    "time": "1 min ago",
                    "userName": "Eu",
                  },
                ]
              }
            >
              <chatItem
                displayPortraitLeft={false}
                message="Test"
                portrait={null}
                time="2 min ago"
                userName="Eu"
              />
              <chatItem
                displayPortraitLeft={false}
                message="Test"
                portrait={null}
                time="1 min ago"
                userName="Eu"
              />
            </loader>
          </div>,
          <chatForm
            isDisabled={false}
          />,
        ],
        "icon": "comments",
        "title": "Chat",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <loader
              loaded={
                Array [
                  Object {
                    "displayPortraitLeft": false,
                    "message": "Test",
                    "portrait": null,
                    "time": "2 min ago",
                    "userName": "Eu",
                  },
                  Object {
                    "displayPortraitLeft": false,
                    "message": "Test",
                    "portrait": null,
                    "time": "1 min ago",
                    "userName": "Eu",
                  },
                ]
              }
            >
              <chatItem
                displayPortraitLeft={false}
                message="Test"
                portrait={null}
                time="2 min ago"
                userName="Eu"
              />
              <chatItem
                displayPortraitLeft={false}
                message="Test"
                portrait={null}
                time="1 min ago"
                userName="Eu"
              />
            </loader>,
            "className": "chat",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <chatItem
                  displayPortraitLeft={false}
                  message="Test"
                  portrait={null}
                  time="2 min ago"
                  userName="Eu"
                />,
                <chatItem
                  displayPortraitLeft={false}
                  message="Test"
                  portrait={null}
                  time="1 min ago"
                  userName="Eu"
                />,
              ],
              "loaded": Array [
                Object {
                  "displayPortraitLeft": false,
                  "message": "Test",
                  "portrait": null,
                  "time": "2 min ago",
                  "userName": "Eu",
                },
                Object {
                  "displayPortraitLeft": false,
                  "message": "Test",
                  "portrait": null,
                  "time": "1 min ago",
                  "userName": "Eu",
                },
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "function",
                "props": Object {
                  "displayPortraitLeft": false,
                  "message": "Test",
                  "portrait": null,
                  "time": "2 min ago",
                  "userName": "Eu",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "function",
                "props": Object {
                  "displayPortraitLeft": false,
                  "message": "Test",
                  "portrait": null,
                  "time": "1 min ago",
                  "userName": "Eu",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "inputValue": undefined,
            "isDisabled": false,
            "onFormSubmit": undefined,
            "onInputChange": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
